sycl/plugins/unified_runtime/ur/adapters/hip/kernel.hpp

```diff
@@ -205,4 +214,16 @@ struct ur_kernel_handle_t_ {
   uint32_t getLocalSize() const noexcept { return Args.getLocalSize(); }
 
   void clearLocalSize() { Args.clearLocalSize(); }
+
+  void addMemObjArg(int Index, ur_mem_handle_t hMem, ur_mem_flags_t Flags) {
+    assert(hMem && "Invalid mem handle");
+    for (auto i = 0u; i < Args.MemObjArgs.size(); ++i) {
+      if (Args.MemObjArgs[i].Index == Index) {
+        // Overwrite the mem obj with the same index
+        Args.MemObjArgs[i] = arguments::mem_obj_arg{hMem, Index, Flags};
+        return;
+      }
+    }
+    Args.MemObjArgs.push_back(arguments::mem_obj_arg{hMem, Index, Flags});
```

Comment:
This does use Index?

Reply:
The difference is that the `Index` is not the same index in the vec, otherwise we would have lots of `nullptr` args in our vec which isn't a good idea IMO. Instead the Index is part of the `mem_obj_arg` struct. Adding a new arg with the same index will successfully overwrite one with the same kernel Index.